crates/stackable-operator/src/builder/pod/container.rs

```diff
@@ -265,7 +309,11 @@ impl ContainerBuilder {
             resources: self.resources.clone(),
             name: self.name.clone(),
             ports: self.container_ports.clone(),
-            volume_mounts: self.volume_mounts.clone(),
+            volume_mounts: if self.volume_mounts.is_empty() {
+                None
+            } else {
+                Some(self.volume_mounts.clone().into_values().collect())
+            },
```

Comment:
note: Inlining this looks kinda convoluted. Please move this out of the struct creation. Also, is there a way to do this with combinators?

Reply:
Moved in b7fcdaa
I could offer the following code, but IMHO it's not really better, WDYT?
```rust
        let volume_mounts = self
            .volume_mounts
            .is_empty()
            .then(|| self.volume_mounts.values().cloned().collect());
```

Reply:
I feel like the combinators are a little better, but I wouldn't block this PR on it.